components/brave_wallet/browser/zcash/zcash_transaction_utils_unittest.cc

```diff
@@ -73,39 +110,190 @@ TEST(ZCashTransactionUtilsUnitTest, PickZCashTransparentInputs) {
+  }
+
+  // Change is 0, but amount is not max.
   {
     std::vector<OrchardNote> notes;
     notes.push_back(OrchardNote{{}, 1u, {}, 100000u, 0, {}, {}});
     notes.push_back(OrchardNote{{}, 2u, {}, 200000u, 0, {}, {}});
     notes.push_back(OrchardNote{{}, 3u, {}, 70000u, 0, {}, {}});
-    auto result = PickZCashOrchardInputs(notes, 150000u);
+    auto result = PickZCashOrchardInputs(notes, 370000u - 15000u,
+                                         ZCashTargetOutputType::kOrchard);
     EXPECT_TRUE(result.has_value());
-    EXPECT_EQ(result->change, 170000u - 150000u - result->fee);  // 17 - 15
+
+    EXPECT_EQ(result->change, 0u);
+    EXPECT_EQ(result->inputs.size(), 3u);
+    // max(2, max(0, 0) + max(3, 1, 2)) * 5000.
+    EXPECT_EQ(result->fee, 15000u);
+
+    EXPECT_EQ(result->inputs[0].amount, 70000u);
+    EXPECT_EQ(result->inputs[0].block_id, 3u);
+    EXPECT_EQ(result->inputs[1].amount, 100000u);
+    EXPECT_EQ(result->inputs[1].block_id, 1u);
+    EXPECT_EQ(result->inputs[2].amount, 200000u);
+    EXPECT_EQ(result->inputs[2].block_id, 2u);
+  }
+
+  // Change is 0, but amount is not max, transparent output.
+  {
+    std::vector<OrchardNote> notes;
+    notes.push_back(OrchardNote{{}, 1u, {}, 100000u, 0, {}, {}});
+    notes.push_back(OrchardNote{{}, 2u, {}, 200000u, 0, {}, {}});
+    notes.push_back(OrchardNote{{}, 3u, {}, 70000u, 0, {}, {}});
+    auto result = PickZCashOrchardInputs(notes, 370000u - 20000u,
+                                         ZCashTargetOutputType::kTransparent);
+    EXPECT_TRUE(result.has_value());
+
+    EXPECT_EQ(result->change, 0u);
+    EXPECT_EQ(result->inputs.size(), 3u);
+    // max(2, max(0, 1) + max(3, 0, 2)) * 5000.
+    EXPECT_EQ(result->fee, 20000u);
+
+    EXPECT_EQ(result->inputs[0].amount, 70000u);
+    EXPECT_EQ(result->inputs[0].block_id, 3u);
+    EXPECT_EQ(result->inputs[1].amount, 100000u);
+    EXPECT_EQ(result->inputs[1].block_id, 1u);
+    EXPECT_EQ(result->inputs[2].amount, 200000u);
+    EXPECT_EQ(result->inputs[2].block_id, 2u);
+  }
+
+  // Transparent output.
+  {
+    std::vector<OrchardNote> notes;
+    notes.push_back(OrchardNote{{}, 1u, {}, 100000u, 0, {}, {}});
+    notes.push_back(OrchardNote{{}, 2u, {}, 200000u, 0, {}, {}});
+    notes.push_back(OrchardNote{{}, 3u, {}, 70000u, 0, {}, {}});
+
+    auto result = PickZCashOrchardInputs(notes, 150000u,
+                                         ZCashTargetOutputType::kTransparent);
+    EXPECT_TRUE(result.has_value());
+    EXPECT_EQ(result->change, 170000u - 150000u - result->fee);
     EXPECT_EQ(result->inputs.size(), 2u);
+    // max(2, max(0, 1) + max(2, 1, 2)) * 5000.
     EXPECT_EQ(result->fee, 15000u);
+
     EXPECT_EQ(result->inputs.size(), 2u);
     EXPECT_EQ(result->inputs[0].amount, 70000u);
     EXPECT_EQ(result->inputs[0].block_id, 3u);
     EXPECT_EQ(result->inputs[1].amount, 100000u);
     EXPECT_EQ(result->inputs[1].block_id, 1u);
   }
 
-  // Full amount
+  // Transparent output, full amount.
```

Comment:
Can we add a test with a single Orchard input to verify the minimum 2 actions rule is enforced? I want to confirm that single input single output still results in 2 actions (10000 fee) and understand how this differs from transparent transactions which have no such minimum?

I may just be missing where this is tested elsewhere though